### ISSUE TITLE
Yale Doorman L3 (Doorbell)

### ISSUE THREAD

**Issue by Mornclaur** [hufftheweevil/node-red-contrib-august#2]
Not really an issue, more like a possible enhancement.

I've been using this with Yale Doorman L3 for a a short while and it works great. 
On the the L3 there is a built in doorbell on the lock.
I had Auto-subscribe turned on and suddenly discovered that there was an event every time the doorbell was pressed.
So thought I could use this to send notifications to speakers, but the problem is that the event from the doorbell is exactly the same as the event from closing the door.

I have tried getDetails, but there is absolutely nothing related to the doorbell anywhere in the complete msg object output.
Is the event output from Auto-subscribe removing anything from what it receives from August? 

Just noticed that Yale Doorman L3 is only sold in scandinavia so might be as simple as the august account register the event, but no part of the locks part of the august api has the event trigger information.

**Comment by hufftheweevil** [comment]
I'd say that this is a better question for the underlying API package, but I just remembered that I manage that one too, so not to worry about that!

When events arrive via pubnub in the API, they are passed along to this package untouched. (With the exception that the `lockId` property is added if it doesn't exist - but that's only because sometime it's listed as `lockID`.) Similarly, those event messages are passed untouched as output in this node.

Unfortunately, that doesn't solve your issue. It is most likely related to inconsistent events on the backend due to the Yale/August merge. The country-specific device probably also does not help.

So sorry I couldn't help fix the problem! I am curious though - do you get door open and/or doorbell info in your phone app for the L3?

**Comment by Mornclaur** [comment]
Yes i get push notifications for the doorbell on the phone and the different events, open, close, lock, operator and doorbell are also listed in the activity feed. To me that indicates that there is some unique identifier for the doorbell event, that is why I asked if the events was passed along untouched.

I think I see a possible work around to pick up the doorbell events. By checking if the states in the new event equals the the states in the previous event I should be able to single out the doorbell events as it just returns the current state of the lock.

**Comment by Mornclaur** [comment]
Sorry for reopening this, but this is the only interface i've tried against the august/yale api that have actually picked up the doorbell being pressed. So thought I might see if there is any way I can assist in figuring it out.

Just looked through your august-api repo, and have a question, In definitions i came over this bit;
{
  remoteEvent: 1,
  ...AugustLockStatus
}

{
  // From August
  status: ['unlocked', 'locked'],
  callingUserID: enum['manuallock', 'manualunlock', UserId],
  doorState: ['open', 'closed'],
  // -----------
  // Added in this module
  state: {
    locked: boolean,
    unlocked: boolean,
    open: boolean,
    closed: boolean
  },
  lockID: string
}

If the event from August contains additional information, what would happen to that information? Is it left out?
If it is left out, is there a way to just show the whole event information raw?

Is there anything I can do to assist in figuring out if there is any additional information that is being sent in the event?

**Comment by hufftheweevil** [comment]
No need to apologize, I'm happy to keep the investigation open.

Nothing is left out, there are only those few things added.

Can you include the event that you get when the door closes vs the event you get when the doorbell rings?

**Comment by Mornclaur** [comment]
Door closed looks like this:
payload: object
status: "unlocked"
callingUserID: "manualunlock"
doorState: "closed"
state: object
locked: false
unlocked: true
open: false
closed: true
lockId: ""
command: "event"

Doorbell looks like this:
payload: object
status: "unlocked"
callingUserID: "manualunlock"
doorState: "closed"
state: object
locked: false
unlocked: true
open: false
closed: true
lockId: ""
command: "event"

I did however notice that if I have the Yale Access app open on my phone at the same time the doorbell was pressed, i get an additional event.  This also happens for open/close and lock/unlock.
These appeared at the same time the push notifications on my phone appeared. 

msg : Object
object
topic: " "
payload: object
remoteEvent: 1
status: "kAugLockState_Unlocked"
info: object
action: "status"
startTime: " "
context: object
transactionID: " "
startDate: " "
retryCount: 1
lockType: "lock_version_7"
serialNumber: " "
rssi: -80
wlanRSSI: -67
wlanSNR: 31
duration: 419
lockID: " "
bridgeID: " "
serial: " "
doorState: "kAugDoorState_Closed"
retryCount: 1
totalTime: 438
resultsFromOperationCache: false
state: object
locked: false
unlocked: true
open: false
closed: true
lockID: " "
lockId: ""
command: "event"
